community/index/src/test/java/org/neo4j/index/internal/gbptree/GBPTreeTest.java

```diff
@@ -684,6 +686,75 @@ public void readHeaderMustThrowIfFileDoesNotExist() throws Exception
         }
     }
 
+    @Test
+    public void readHeaderMustThrowIOExceptionIfFileIsEmpty() throws Exception
```

Comment:
Really sweet testing!
Can we also verify that constructor behave the same way as `readHeader` here?

Reply:
Oh of course, good call. I'll duplicate these tests for the constructor!